src/Provider/AbstractProvider.php

```diff
@@ -349,14 +349,20 @@ protected function fetchProviderData($url, array $headers = [])
             $response = $request->getBody();
         } catch (BadResponseException $e) {
             // @codeCoverageIgnoreStart
-            $raw_response = explode("\n", $e->getResponse());
-            throw new IDPException(end($raw_response));
+            $message = $this->getResponseErrorMessage($e->getResponse()->getBody());
+            throw new IDPException($message);
```

Comment:
Yeah. `IDPException` in master is expecting an array, and then it attempts to parse values out of that array. I don't think I can merge this into master, as a result, since it will break other providers' implementations.

Thoughts?